### PR TITLE
An 470/synthetic balances terra

### PR DESCRIPTION
**Pull request by amasucci13**
- Combined synthetic balances with native balances already in gold terra.daily_balances
- Moved de-dupeing logic to silver.synthetic_balances instead of gold. 
-  Added is_native (boolean) column
- Added tests for silver synthetic balances and terra.daily_balances (looking at last 30 days)
- All tests pass except for silver synthetic - recent data. Last date is 12/30 (Looking into root cause).
- Ran full refreshes and incremental refreshes successfully.

### CODE REVIEW

**Review by jkallogjeri — APPROVED** [review]

File: models/terra/silver/silver_terra__block_synthetic_balances.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  chain_id,
+  SUBSTRING(inputs,25,44) as address,
+  b.value:denom::string as currency,
+  'liquid' as balance_type,
+  false as is_native,
+  b.value:amount::FLOAT as balance,
+  row_number() OVER (PARTITION BY date_trunc('day', block_timestamp), address, currency, chain_id ORDER BY block_timestamp DESC, block_id DESC) as rn
+FROM {{ ref('terra_dbt__synthetic_balances') }},
+LATERAL FLATTEN(input => value_obj :balances) b
+WHERE 1 = 1
```
Comment (juls858) [inline models/terra/silver/silver_terra__block_synthetic_balances.sql:25]: this line doesn't do anything

File: models/terra/silver/silver_terra__block_synthetic_balances.sql [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+SELECT * FROM dedupe
+WHERE rn = 1
+
+{% if is_incremental() %}
+AND system_created_at :: DATE >= ( SELECT DATEADD('day', -1, MAX(system_created_at :: DATE)) FROM {{ this }})
```
Comment (juls858) [inline models/terra/silver/silver_terra__block_synthetic_balances.sql:32]: this should be pushed into the inner query for optimization. We should filter fist and then dedupe as it's more efficient to dedupe less rows than more rows.  Also, using the [qualify](https://github.com/FlipsideCrypto/sql_models/blob/a643e98f61e6cd427083cc584b2c4441b835f9e6/models/crosschain/silver/silver_crosschain__address_labels.sql#L33-L35) statement might save you a few lines of code

File: models/terra/silver/silver_terra__block_synthetic_balances.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          combination_of_columns:
+            - BLOCK_TIMESTAMP
+            - ADDRESS
+            - CURRENCY
+          where: BLOCK_TIMESTAMP > CURRENT_DATE - 30 AND BALANCE > 0
+    columns:
+      - name: BLOCK_ID
+        tests:
+          - not_null:
+              where: BLOCK_TIMESTAMP > CURRENT_DATE - 30
```
Comment (juls858) [inline models/terra/silver/silver_terra__block_synthetic_balances.yml:15]: the individual column level test should be fast wtihout the date filter.. can you provide verify how long these individual tests take? If the time is reasonable, a couple of mintes, lets remove the date filter. 

The unique key test is probably the one that takes the longest, so we can keep the date filter there for now

File: models/terra/silver/silver_terra__block_synthetic_balances.sql (hunk 2 above)
Comment (juls858) [reply models/terra/silver/silver_terra__block_synthetic_balances.sql:32]: lets change the incremental window to 2 days just to be safe.

**Review by juls858 — CHANGES_REQUESTED** [review]

**Review by juls858 — APPROVED** [review]

**Pull request merged**